tools/snp.sh

```diff
@@ -910,9 +943,14 @@ setup_guest_attestation() {
   else
     git checkout "current/${SNPGUEST_BRANCH}"
   fi
-
-  cargo build -r
-  scp_guest_command target/release/snpguest "${GUEST_USER}@localhost:/home/${GUEST_USER}"
+  
+  if "$SVSM"; then
+    cargo build --release --target x86_64-unknown-linux-musl
+    scp_guest_command target/x86_64-unknown-linux-musl/release/snpguest "${GUEST_USER}@localhost:/home/${GUEST_USER}"
+  else
+    cargo build -r
+    scp_guest_command target/release/snpguest "${GUEST_USER}@localhost:/home/${GUEST_USER}"
+  fi
```

Comment:
@larrydewey & @ryansavino Not the biggest deal, but Rama had to add this due to dependency differences in Ubuntu 22.04 and 24.04. Do we want to default to musl builds? If I'm not mistaken this will happen to anyone that has a 24.04 host that tries to use the script to launch a 22.04 guest. It's unrelated to svsm.

Reply:
I agree. We should probably be using the musl binary when building for something like this to remove failure layers where possible.